src/main/kotlin/cacophony/codegen/patterns/cacophonyPatterns/ValuePatterns.kt

```diff
@@ -5,131 +5,136 @@ import cacophony.codegen.patterns.SlotFill
 import cacophony.codegen.patterns.ValuePattern
 import cacophony.controlflow.*
 
-class ValuePatterns {
-    abstract class BinaryOpPattern {
-        val lhsLabel = ValueLabel()
-        val rhsLabel = ValueLabel()
-        val lhsSlot = CFGNode.ValueSlot(lhsLabel)
-        val rhsSlot = CFGNode.ValueSlot(rhsLabel)
-    }
-
-    abstract class UnaryOpPattern {
-        val childLabel = ValueLabel()
-    }
-
-    // for now, we can always dump a constant into a register and call it a value
-    object ConstantPattern : ValuePattern {
-        val label = ConstantLabel()
-        val predicate: (Int) -> Boolean = { _ -> true }
-
-        override val tree = CFGNode.ConstantSlot(label, predicate)
-
-        override fun makeInstance(fill: SlotFill, destination: Register): List<Instruction> {
-            TODO("Not yet implemented")
-        }
+val valuePatterns =
+    listOf(
+        ConstantPattern,
+        AdditionPattern,
+        SubtractionPattern,
+        MultiplicationPattern,
+        DivisionPattern,
+        ModuloPattern,
+        EqualsValuePattern,
+        NotEqualsValuePattern,
+        LessValuePattern,
+        GreaterValuePattern,
+        LessEqualValuePattern,
+        GreaterEqualValuePattern,
```

Comment:
NIT: In `ConditionPatterns.kt`, these are listed in a different order: lt, leq, gt, geq. I think it would be nice to keep those orders aligned. Personally, I like the order from `ConditionPatterns.kt` more, but it's not a strong opinion